### PR TITLE
Upgrade CI

### LINKED ISSUE #40: CI using unmaintained components
The GH-actions in [action-rs](https://github.com/actions-rs) are no longer maintained and should be replaced with something which an active maintainer.

### PR DESCRIPTION
**Pull request by simonsso**
Remove unsupported versions

close #40

### CODE REVIEW

**Comment by simonsso** [comment]
@maroux ping

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -16,38 +16,30 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        rust_toolchain: [1.64.0, ""] # MSRV + ("" = rust-toolchain version)
+        rust_toolchain: [1.70.0, "stable"] # MSRV + ("" = rust-toolchain version)
```
Comment (rnarubin) [inline .github/workflows/ci.yml:19]: does the new action support 1.64.0? An msrv bump seems orthogonal, unless it's necessary

**Comment by rnarubin** [comment]
Replacing the archived actions-rs tools is a good idea, thanks for the change. I'd prefer to keep tooling changes separate from policy changes, see my comment inline

**Comment by simonsso** [comment]
> Replacing the archived actions-rs tools is a good idea, thanks for the change. I'd prefer to keep tooling changes separate from policy changes, see my comment inline

Merge #39 first, as some of the dependencies have bumped their MSRV

File: .github/workflows/ci.yml (hunk 1 above)
Comment (simonsso) [reply .github/workflows/ci.yml:19]: > does the new action support 1.64.0? An msrv bump seems orthogonal, unless it's necessary

Running a `cargo update` breaks the build if not incremented.

Comment (simonsso) [reply .github/workflows/ci.yml:19]:
```suggestion
        rust_toolchain: [1.64.0, "stable"] 
```

**Comment by simonsso** [comment]
> Replacing the archived actions-rs tools is a good idea, thanks for the change. I'd prefer to keep tooling changes separate from policy changes, see my comment inline

Looks good now:

https://github.com/simonsso/ya-gcp/actions/runs/8074207307/job/22059134224?pr=3

**Comment by simonsso** [comment]
@rnarubin you can modify this as you like

**Comment by simonsso** [comment]
@rnarubin  This is updated as you requested and is verified to build on my github repo. Please merge/edit or reject this PR.
